File: server/s2game/src/main/java/com/spectopics/s2game/dto/serverPayloads/ResultsStageStartedPayload.java
package com.spectopics.s2game.dto.serverPayloads;

import com.spectopics.s2game.models.LobbyState;

public class ResultsStageStartedPayload {
    // Battle over, tells all clients the results stage has started, and gives them the current state of the lobby.
    public LobbyState lobbyState;

    public ResultsStageStartedPayload(LobbyState lobbyState) {
        this.lobbyState = lobbyState;
    }
}

File: front-end/src/dto/serverPayloads/ResultsStageStartedPayload.js
import { LobbyState } from '../../models/LobbyState.js';

export class ResultsStageStartedPayload {
    constructor(lobbyState) {
        this.lobbyState = lobbyState;
    }

    static fromDict(data) {
        return new ResultsStageStartedPayload(LobbyState.fromDict(data.lobbyState));
    }
}
